app/controllers/users_controller.rb

```diff
@@ -0,0 +1,42 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      redirect_to root_url, notice: 'You have signed up succesfully, please sign in'
+    else
+      render 'new', alert: 'Sign up failed, Please check and try again'
+    end
+  end
+
+  def show
+    @user = User.find(params[:id])
+    @upcoming_events = @user.created_events.upcoming_events
+    @previous_events = @user.created_events.previous_events
+    @previously_attended_events = current_user.attended_events.previous_events
+    @to_attend_events = current_user.attended_events.upcoming_events
+  end
```

Comment:
- [ ]  lists events created by user, i.e. @created_events = current_user.events
Note : Just list events created by the user and user `current_user` helper